tests/components/mobile_app/test_http_api.py

```diff
@@ -100,3 +105,48 @@ async def test_registration_encryption(hass, hass_client):
     decrypted_data = decrypted_data.decode("utf-8")
 
     assert json.loads(decrypted_data) == {"one": "Hello world"}
+
+
+async def test_duplicate_device_name(hass, hass_client, hass_admin_user):
+    """Test that registering duplicate device name adds suffix."""
+    await async_setup_component(hass, DOMAIN, {DOMAIN: {}})
+
+    api_client = await hass_client()
+
+    # Register first device
+    with patch(
+        "homeassistant.components.person.async_add_user_device_tracker",
+        spec=True,
+        return_value=mock_coro(),
```

Comment:
`mock_coro` is legacy. Please import `patch` from `tests.async_mock`. That patch function will handle patching coroutines and we can set return values for normal functions.

Reply:
Done, also replaced elsewhere in that test set